engine-tests/src/test/java/org/terasology/config/flexible/validators/RangedNumberValidatorTest.java

```diff
@@ -0,0 +1,210 @@
+
+            for (int i = validator.getMin(); i <= validator.getMax(); i++) {
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
+
+            assertFalse(validator.fastValidate(validator.getMin() - 1));
+            assertFalse(validator.fastValidate(validator.getMax() + 1));
+        }
+
+        @Test
+        public void testMinExclusive() {
+            initValidator(0, 100, false, true);
+
+            for (int i = validator.getMin() + 1; i < validator.getMax(); i++) {
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
+
+            assertFalse(validator.fastValidate(validator.getMin()));
+            assertTrue(validator.fastValidate(validator.getMax()));
+        }
+
+        @Test
+        public void testMaxExclusive() {
+            initValidator(0, 100, true, false);
+
+            for (int i = validator.getMin() + 1; i < validator.getMax(); i++) {
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
+
+            assertTrue(validator.fastValidate(validator.getMin()));
+            assertFalse(validator.fastValidate(validator.getMax()));
+        }
+
+        @Test
+        public void testAllExclusive() {
+            initValidator(0, 100, false, false);
+
+            for (int i = validator.getMin() + 1; i < validator.getMax(); i++) {
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
+
+            assertFalse(validator.fastValidate(validator.getMin()));
+            assertFalse(validator.fastValidate(validator.getMax()));
+        }
+
+        @Test
+        public void testLowUnbounded() {
+            initValidator(null, 100, false, false);
+
+            assertTrue(validator.fastValidate(-1000));
+            assertTrue(validator.fastValidate(-50000));
+
+            assertTrue(validator.fastValidate(50));
+
+            assertFalse(validator.fastValidate(validator.getMax() + 1));
+        }
+
+        @Test
+        public void testHighUnbounded() {
+            initValidator(0, null, false, false);
+
+            assertTrue(validator.fastValidate(1000));
+            assertTrue(validator.fastValidate(50000));
+
+            assertTrue(validator.fastValidate(50));
+
+            assertFalse(validator.fastValidate(validator.getMin() - 1));
+        }
+
+        @Test
+        public void testAllUnbounded() {
+            initValidator(null, null, false, false);
+
+            assertTrue(validator.fastValidate(1000));
+            assertTrue(validator.fastValidate(50000));
+            assertTrue(validator.fastValidate(50));
+
+            assertTrue(validator.fastValidate(-1000));
+            assertTrue(validator.fastValidate(-50000));
+        }
+    }
+
+    public static class DoubleValidator {
+        private static final double EPSILON = 0.000001d;
```

Comment:
Epsilon is indeed often a synonym with "error", but I'd name it more eplicitly, i.e. ALLOWED_ERROR, for the benefit of younger students, i.e. GCI, which may or may not have encountered yet that notion.